core/startup/config_file_watcher.go

```diff
@@ -15,7 +15,7 @@ import (
 type WatchConfigDirectoryCloser func() error
 
 func ReadApiKeysJson(configDir string, appConfig *config.ApplicationConfig) error {
-	fileContent, err := os.ReadFile(path.Join(configDir, "api_keys.json"))
+	fileContent, err := os.ReadFile(filepath.Join(configDir, "api_keys.json"))
 	if err == nil {

```

Comment:
one of the errors that is likely to happen is the file not existing, we should catch that with something like:
```
	if err != nil {
		if os.IsNotExist(err) {
			return nil
		}
		return err
	}
```